examples/azure/terraform-azure-nsg-example/main.tf

```diff
@@ -0,0 +1,134 @@
+provider "azurerm" {
+  version = "~>2.20"
+  features {}
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# PIN TERRAFORM VERSION TO >= 0.12
+# The examples have been upgraded to 0.12 syntax
+# ---------------------------------------------------------------------------------------------------------------------
+
+terraform {
+  required_version = ">= 0.12"
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY A RESOURCE GROUP
+# See test/terraform_azure_example_test.go for how to write automated tests for this code.
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "azurerm_resource_group" "main" {
+  name     = "${var.resource_group_name}-${var.postfix}"
+  location = var.location
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY VIRTUAL NETWORK RESOURCES
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "azurerm_virtual_network" "main" {
+  name                = "${var.vnet_name}-${var.postfix}"
+  address_space       = ["10.0.0.0/16"]
+  location            = azurerm_resource_group.main.location
+  resource_group_name = azurerm_resource_group.main.name
+}
+
+resource "azurerm_subnet" "internal" {
+  name                 = "${var.subnet_name}-${var.postfix}"
+  resource_group_name  = azurerm_resource_group.main.name
+  virtual_network_name = azurerm_virtual_network.main.name
+  address_prefixes       = ["10.0.17.0/24"]
+}
+
+resource "azurerm_network_interface" "main" {
+  name                = "${var.vm_nic_name}-${var.postfix}"
+  location            = azurerm_resource_group.main.location
+  resource_group_name = azurerm_resource_group.main.name
+
+  ip_configuration {
+    name                          = "${var.vm_nic_ip_config_name}-${var.postfix}"
+    subnet_id                     = azurerm_subnet.internal.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_network_security_group" "main" {
```

Comment:
suggest changing resource name to nsg or nsg_example

Reply:
Updated to nsg_example